binding.cc

```diff
@@ -56,13 +57,38 @@ typedef struct {
   bool valid;
 } bare_sdl_audio_device_format_t;
 
+typedef struct {
+  SDL_Camera *handle;
+} bare_sdl_camera_t;
+
+enum bare_camera_format_status : uint8_t {
+  BARE_CAMERA_FORMAT_OK = 0,
+  BARE_CAMERA_FORMAT_PENDING = 1,
+  BARE_CAMERA_FORMAT_ERROR = 2
+};
+
+typedef struct {
+  SDL_CameraSpec spec;
+  bare_camera_format_status status;
+} bare_sdl_camera_spec_t;
+
+typedef struct {
+  SDL_CameraSpec spec;
+  bare_camera_format_status status;
+} bare_sdl_camera_format_t;
```

Comment:
I spotted a crash because `bare_sdl_camera_spec_t` where used in a `bare_sdl_camera_format_t` API.

I added the `status` field as bandage, but I would to let you go through it another time.

> Note: @kasperisager mentioned the [`SDL_GetCameraPermissionState`](https://wiki.libsdl.org/SDL3/SDL_GetCameraPermissionState) if it could help